movai_core_shared/core/message_client.py

```diff
@@ -125,3 +138,64 @@ async def send_msg(self, data: dict, **kwargs) -> None:
         msg.update(kwargs)
 
         self._zmq_client.send(msg)
+
+
+class AsyncMessageClient(MessageClient):
+    def _set_zmq_client(self, identity: str):
+        self._zmq_client = AsyncZMQClient(identity, self._server_addr)
+
+    async def send_request(
```

Comment:
isn't this a duplicated method ? if yes can't we do inheritance to base class implementing common behaviour ?

Reply:
that's what we do, we inherit from MessageClient class and change implementation for async ones